File: com.dexaint.dukechess/src/main/java/com/dexaint/dukechess/movement/Command.java
package com.dexaint.dukechess.movement;

import org.apache.commons.lang3.ArrayUtils;

import com.dexaint.dukechess.flow.Field;
import com.dexaint.dukechess.flow.Player;

public class Command extends MovementImpl {
	public int[] validateMovement(Field field, int pos, Destination d, Player p) {
		int[] ret = new int[]{};
		
		ret = ArrayUtils.add(ret, offset2Dest(field, pos, d.dest2Offset(p)));
		
		return ret;
	}
}

File: deprecated/Unity/Assets/Scripts/Movement/Command.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Command : Movement
{
    public override List<int> ValidateMovement(List<ChessController> board, int pos, DestinationParser.Destination d, Player p)
    {
        List<int> ret = new List<int>();

        int[] offsets = DestinationParser.Dest2Offset(d, p);

        if (!IsInField(pos, offsets)) return ret;

        ret.Add(Offset2Dest(pos, DestinationParser.Dest2Offset(d, p)));

        return ret;
    }
}
